internal/srv/app.go

```diff
@@ -43,8 +45,13 @@ func (s *Server) removeNamespace(ns string) error {
 
 // CreateNamespace creates namespaces for the specified group that is
 // provided in the event received
-func (s *Server) CreateNamespace(groupID string) (*v1.Namespace, error) {
-	s.Logger.Debugf("ensuring namespace %s exists", groupID)
+func (s *Server) CreateNamespace(name string, hash string) (*v1.Namespace, error) {
```

Comment:
Sadly, at this point, yes. The problem that I was dealing with is that the newly generated ID's can contain a mixture of special characters and upper/lower that was causing problems with the dns style limitation on naming kube objects. I add the actual name to the namespace as an annotation, so that'll give us a way of actually searching for things we want (and I imagine we'll stuff more things in there like this as needed).

Went with a hash because it naturally gave us the naming conventions I was looking for vs a chain of string filters and base64'ing my way through. But I also don't have strong opinions on if folks would prefer something more in that vein.